crates/hyperswitch_domain_models/src/merchant_connector_account.rs

```diff
@@ -530,31 +559,74 @@ impl From<MerchantConnectorAccountUpdate> for MerchantConnectorAccountUpdateInte
-
-impl MerchantConnectorAccounts {
-    pub fn new(merchant_connector_accounts: Vec<MerchantConnectorAccount>) -> Self {
-        Self(merchant_connector_accounts)
-    }
-
-    pub fn is_merchant_connector_account_id_in_connector_mandate_details(
-        &self,
-        profile_id: Option<&id_type::ProfileId>,
-        connector_mandate_details: &diesel_models::PaymentsMandateReference,
-    ) -> bool {
-        let mca_ids = self
-            .0
-            .iter()
-            .filter(|mca| {
-                mca.disabled.is_some_and(|disabled| !disabled)
-                    && profile_id.is_some_and(|profile_id| *profile_id == mca.profile_id)
-            })
-            .map(|mca| mca.get_id())
-            .collect::<std::collections::HashSet<_>>();
+common_utils::create_list_wrapper!(
+    MerchantConnectorAccounts,
+    MerchantConnectorAccount,
+    impl_functions: {
+        #[cfg(feature = "v2")]
+        pub fn get_connector_and_supporting_payment_method_type_for_session_call(
+            &self,
+        ) -> Vec<(&MerchantConnectorAccount, common_enums::PaymentMethodType)> {
+            let mut connector_and_supporting_payment_method_type = Vec::new();
+            self.iter().for_each(|connector_account| {
+                let res = connector_account
+                    .get_parsed_payment_methods_enabled()
+                    .into_iter()
+                    .filter_map(|parsed_payment_method_result| {
+                        parsed_payment_method_result
+                            .inspect_err(|err| {
+                                logger::error!(session_token_parsing_error=?err);
+                            })
+                            .ok()
+                    })
+                    .flat_map(|parsed_payment_methods_enabled| {
+                        parsed_payment_methods_enabled
+                            .payment_method_types
+                            .unwrap_or_default()
+                            .into_iter()
+                            .filter(|payment_method_type| {
+                                let is_invoke_sdk_client = matches!(
+                                    payment_method_type.payment_experience,
+                                    Some(api_models::enums::PaymentExperience::InvokeSdkClient)
+                                );
+                                is_invoke_sdk_client
+                            })
+                            .map(|payment_method_type| {
+                                (connector_account, payment_method_type.payment_method_type)
+                            })
+                            .collect::<Vec<_>>()
+                    })
+                    .collect::<Vec<_>>();
+                connector_and_supporting_payment_method_type.extend(res);
+            });
+            connector_and_supporting_payment_method_type
+        }
+        pub fn filter_based_on_profile_and_connector_type(
```

Comment:
can we have this filter on the query itself? cc: @jarnura

Reply:
Will take this up in a new PR.

Reply:
we can filter on the query, since these are domain models and the function present in domain model crate